src/components/numeral-date/numeral-date.pw.tsx

```diff
@@ -306,6 +307,132 @@ test.describe("NumeralDate component", () => {
     }
   );
 
+  ([
+    ["02", "29", "2001", "Day in February should be a number within 1-28."],
+    ["02", "30", "2004", "Day in February should be a number within 1-29."],
+    ["06", "31", "2001", "Day in June should be a number within 1-30."],
+    ["09", "31", "2001", "Day in September should be a number within 1-30."],
+    ["12", "32", "2001", "Day in December should be a number within 1-31."],
```

Comment:
**suggestion(non-blocking):** Reckon it's worth moving in to a `const`? I just think that if we need to add anymore test cases in to this, we'll only have to update it one place instead of four.